model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -561,6 +568,36 @@ object WorkspaceState {
   case object DeleteFailed extends WorkspaceState
 }
 
+case class WorkspaceSetting(`type`: WorkspaceSettingType, config: WorkspaceSettingConfig)
+
+object WorkspaceSettingTypes {
+  sealed trait WorkspaceSettingType extends RawlsEnumeration[WorkspaceSettingType] {
+    override def toString: String = getClass.getSimpleName.stripSuffix("$")
+    override def withName(name: String): WorkspaceSettingType = WorkspaceSettingTypes.withName(name)
+    def defaultConfig(): WorkspaceSettingConfig
+  }
+
+  def withName(name: String): WorkspaceSettingType = name.toLowerCase match {
+    case "gcpbucketlifecycle" => GcpBucketLifecycle
+    case _                    => throw new RawlsException(s"invalid WorkspaceSetting [$name]")
+  }
+
+  case object GcpBucketLifecycle extends WorkspaceSettingType {
+    override def defaultConfig(): WorkspaceSettingConfig = GcpBucketLifecycleConfig(List.empty)
+  }
+}
+
+sealed trait WorkspaceSettingConfig
+object WorkspaceSettingConfig {
+  case class GcpBucketLifecycleConfig(rules: List[GcpBucketLifecycleRule]) extends WorkspaceSettingConfig
+
+  case class GcpBucketLifecycleRule(action: GcpBucketLifecycleAction, conditions: GcpBucketLifecycleCondition)
+  case class GcpBucketLifecycleAction(`type`: String)
```

Comment:
That's fair. I avoided it initially with `WorkspaceSetting` because I'm also not a fan, but then I wanted this to mimic the Google lifecycle rules format as closely as possible so I ended up caving and using backticks for both since they use `type`.